### ISSUE TITLE
SD reproducibility

### ISSUE THREAD

**Issue by neudachina** [vinhsuhi/LD3#4]
Dear authors,

We faced the problems with reproducing of SD FID values reported in your paper

You write about having 
- `FID=18.05` for `NFE=4`
- `FID=14.11` for `NFE=5`
- `FID=12.10` for `NFE=6`
- `FID=11.80` for `NFE=7`

using iPNDM solver with GITS timesteps, whilst we observe 

- `FID=18.80` for `NFE=4`
- `FID=14.99` for `NFE=5`
- `FID=13.12` for `NFE=6`
- `FID=12.61` for `NFE=7`

We calculate reference statistics using COCO_val2014 dataset that consists of 40504 images and we use the following script:
```
python3 gen_data.py \
--all_config configs/stable_diff_v1-5.yml \
--total_samples 30000 \
--num_prompts 30000 \
--num_samples_per_prompt 1 \
--sampling_batch_size 16 \
--steps $N \
--solver_name ipndm \
--skip_type gits \
--use_gits \
--save_png --data_dir train_data_test2/train_data_stable_diff_v1-5 \
--prompt_path data/coco_captions.txt
```
`data/coco_captions.txt` is the same file you have in you repo. To calculate statistics and final FID numbers, we use pytorch-fid.

Is there something we are missing in reproducing the results? 
We observed that the FID metric of NFE=4 stays the same for different seeds, so it shouldn't be the cause

Thank you for you answer!

**Comment by hoangdzung** [comment]
Thank you for reaching out.

Regarding your question, we compute the FID scores in our experiments using only 30,000 validation images corresponding to 30,000 captions. Our calculated reference statistics are [here](https://drive.google.com/file/d/1hlSlPbg1ycsPq7ZnYHBooVWuIiu3mAxe/view?usp=sharing).

We also tested using FID with the full validation set (40,504 images) but could not reproduce your reported FID values.

Additionally, we have added a compute_fid script to our repository, which combines the gen_data and compute_fid scripts into a single streamlined workflow. You can use this for easier reproduction.

Please let us know if you have any further questions.
